cdap-security/src/main/java/co/cask/cdap/security/store/secretmanager/SecretManagerSecureStoreService.java

```diff
@@ -0,0 +1,200 @@
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.common.namespace.NamespaceQueryAdmin;
+import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.proto.id.SecureKeyId;
+import co.cask.cdap.securestore.spi.SecretManager;
+import co.cask.cdap.securestore.spi.SecretManagerContext;
+import co.cask.cdap.securestore.spi.SecretNotFoundException;
+import co.cask.cdap.securestore.spi.secret.Secret;
+import co.cask.cdap.securestore.spi.secret.SecretMetadata;
+import co.cask.cdap.security.store.SecureStoreService;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.util.concurrent.AbstractIdleService;
+import com.google.inject.Inject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Secret manager secure store service to initialize {@link SecretManager}.
+ */
+public class SecretManagerSecureStoreService extends AbstractIdleService implements SecureStoreService {
+  private static final Logger LOG = LoggerFactory.getLogger(SecretManagerSecureStoreService.class);
+
+  private final NamespaceQueryAdmin namespaceQueryAdmin;
+  private final SecretManagerContext context;
+  private final String type;
+  private SecretManager secretManager;
+
+  @Inject
+  public SecretManagerSecureStoreService(CConfiguration cConf, NamespaceQueryAdmin namespaceQueryAdmin) {
+    this.namespaceQueryAdmin = namespaceQueryAdmin;
+    this.context = new DefaultSecretManagerContext(cConf);
+    this.type = cConf.get(Constants.Security.Store.PROVIDER);
+    this.secretManager = new SecretManagerExtensionLoader(cConf.get(Constants.Security.Store.EXTENSIONS_DIR)).get(type);
+  }
+
+  @VisibleForTesting
+  public SecretManagerSecureStoreService(NamespaceQueryAdmin namespaceQueryAdmin, SecretManagerContext context,
+                                         String type, SecretManager secretManager) {
+    this.namespaceQueryAdmin = namespaceQueryAdmin;
+    this.context = context;
+    this.type = type;
+    this.secretManager = secretManager;
+  }
+
+  @Override
+  protected void startUp() throws Exception {
+    LOG.info("Starting {}.", getClass().getSimpleName());
+    initializeSecretManager();
+  }
+
+  @Override
+  protected void shutDown() throws Exception {
+    LOG.info("Stopping {}.", getClass().getSimpleName());
+    destroySecretManager();
+  }
+
+  /**
+   * List of all the secrets in the secret manager belonging to the specified namespace.
+   * @return A map of secret name to its description
+   * @param namespace the namespace for list of secrets
+   * @throws NamespaceNotFoundException if the specified namespace does not exist
+   * @throws IOException if there was a problem while reading from the secret manager
+   */
+  @Override
+  public Map<String, String> listSecureData(String namespace) throws Exception {
```

Comment:
This API is wrong https://issues.cask.co/browse/CDAP-13648

Please remember to fix that bug.

Reply:
yea will fix it in another PR.